### PR TITLE
Fix/upload file directly to entity

### PR DESCRIPTION
**Pull request by petersg83**
fix https://github.com/strapi/strapi/issues/12673

### CODE REVIEW

File: packages/core/upload/server/services/upload.js [hunk 1]
```diff
@@ -45,6 +45,20 @@ const sendMediaMetrics = data => {
   }
 };
 
+const addTmpWorkingDirectory = async (files, providedTmpWorkingDirectory) => {
```
Comment (gu-stav) [inline packages/core/upload/server/services/upload.js:48]: Do you need the `providedTmpWorkingDirectory` argument? This PR doesn't seem using it anywhere.

I'd call this method `createAndAssignTmpWorkingDirectoryToFiles` instead. I think it needs to be clear the methods actually mutates the files too. Wdyt?

**Review by gu-stav — COMMENTED** [review]
Two comments about the interface & name of the method. The rest looks really good to me. 🌞

**Review by gu-stav — APPROVED** [review]
Good job. Thanks for fixing the issue! 🚀

**Review by MarionLemaire — APPROVED** [review]
Checked on Ubuntu using Chrome with PNG and JPG files : OK
curl --request POST --url http://localhost:1337/api/dogs --header 'Content-Type: multipart/form-data' --form 'files.DogPic=@/home/yon/DSC_3918.JPG' --form 'data={}'

**Pull request merged**